src/shared/contexts/pinneditems.tsx

```diff
@@ -0,0 +1,151 @@
+  updatedAt: string
+  metadata: {
+    [key: string]: string
+  }
+  type: string
+}
+
+export enum PinnedItemTypes {
+  Dashboard = 'dashboard',
+  Cell = 'cell',
+  Notebook = 'notebook',
+  Task = 'task',
+  Alert = 'alert',
+  UploadCSV = 'uploadCsv',
+  UploadLP = 'uploadLP',
+  ClientLibrary = 'clientLibrary',
+}
+export const PinnedItemsContext = createContext(null)
+
+// The routes below are placeholders until we get the implementation of the swagger-defined routes through the pipeline
+
+export const getPinnedItems = async () => {
+  const pinnedRes = await getPinned({})
+  if (pinnedRes.status !== 200) {
+    throw new Error('Unable to retrieve pinned items')
+  } else {
+    return pinnedRes.data
+  }
+}
+
+const removePinnedItem = async (id: string) => {
+  await deletePinned({id})
+}
+const addPinnedItem = async (item: Partial<PinnedItem>) => {
+  const pinnedRes = await postPinned({data: item})
+  if (pinnedRes.status !== 200) {
+    throw new Error('Unable to retrieve pinned items')
+  } else {
+    return
+  }
+}
+
+const updatePinnedItem = async (id: string, item: Partial<PinnedItem>) => {
+  await putPinned({id, data: item})
+
+  return
+}
+
+export const pushPinnedItem = async (newItem: Partial<PinnedItem>) => {
+  try {
+    await addPinnedItem(newItem)
+  } catch (err) {
+    console.error(err)
+  }
+}
+
+export const deletePinnedItemByParam = async (param: string) => {
+  try {
+    const items = await getPinnedItems()
+    const toDeleteItem = items.find(item =>
+      Object.values(item.metadata).includes(param)
+    )
+
+    if (toDeleteItem) {
+      await removePinnedItem(toDeleteItem.id)
+    }
+  } catch (err) {
+    console.error(err)
+  }
+}
+
+export const deletePinnedItem = async (itemID: string) => {
+  try {
+    await removePinnedItem(itemID)
+  } catch (err) {
+    console.error(err)
+  }
+}
+
+export const updatePinnedItemByParam = async (id: string, updateParams: {}) => {
+  try {
+    const pinnedItems = await getPinnedItems()
+    const toUpdateItem = pinnedItems?.find(item =>
+      Object.values(item.metadata).includes(id)
+    )
+    if (toUpdateItem) {
+      await updatePinnedItem(toUpdateItem.id, {
+        metadata: {...toUpdateItem.metadata, ...updateParams},
+      })
+    }
+  } catch (err) {
+    console.error(err)
+  }
+}
+
+interface Props {
+  children: ReactElement<any>
+}
+
+const PinnedItemsProvider: FC<Props> = ({children}) => {
+  const [pinnedItems, setPinnedItems] = useState([])
+
+  const deletePinnedItemsHelper = useCallback(
+    async (id: string) => {
+      await deletePinnedItem(id)
+      setPinnedItems(pinnedItems.filter(item => item.id !== id))
+    },
+    [pinnedItems]
+  )
+
+  useEffect(() => {
+    getPinnedItems().then(data => setPinnedItems(data))
+  }, [])
+
+  return (
+    <PinnedItemsContext.Provider value={{pinnedItems, deletePinnedItemsHelper}}>
+      {children}
+    </PinnedItemsContext.Provider>
+  )
+}
+
+export default memo(PinnedItemsProvider)
```

Comment:
why are we awaiting all the things?

Reply:
Patience is a virtue

Seriously tho, no particular reason one way or another, notice where i invoke them now, they are not being called synchronously so either way, shouldn't be an issue

Reply:
let's remove it. Feels like something should be done afterwards, but we're not

Reply:
sounds good

Reply:
On second thought, id like to leave the functions async. i still want to have the functions be able to have the error handling that i've left on there, and ultimately this will still run async. I think no harm done, no?